requests/api.py

```diff
@@ -19,7 +19,7 @@ def request(method, url, **kwargs):
     :param method: method for the new :class:`Request` object.
     :param url: URL for the new :class:`Request` object.
     :param params: (optional) Dictionary or bytes to be sent in the query string for the :class:`Request`.
-    :param data: (optional) Dictionary (will be form-encoded), bytes, or file-like object to send in the body of the :class:`Request`.
+    :param data: (optional) Dictionary or list of tuples ``[(key, value)]`` (will be form-encoded), bytes, or file-like object to send in the body of the :class:`Request`. The list of tuple version is to allow duplication of fields.
```

Comment:
I think we'd rather avoid the extra sentence at the end: we don't really need to justify the reasoning for the data types here because it starts drastically increasing the length of the property documentation.

If we want to add stuff about why we allow different formats we should add a new section to the prose documentation about the fact that almost everywhere that accepts a dictionary also accepts an iterable of tuples.